### PR TITLE
Add a search function

### PR DESCRIPTION
**Pull request by jsanz**
Adds a new form input and  `/tile` parameter to pass to Elasticsearch a search query.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -45,7 +45,14 @@ const server = http.createServer(async function (request, response) {
                 grid_type: 'grid',
                 size: params.renderMethod === 'hits' ? 10000 : 0,// only populate the hits layer when necessary
                 track_total_hits: false,
-                body: {}
+                body: params.searchQuery ? {
```
Comment (nreese) [inline index.js]: Now that you are setting body, would it makes sense to move all parameters there?

Comment (jsanz) [reply index.js]: Yep, seems better to put all the keys inside `body` following [the documentation](https://www.elastic.co/guide/en/elasticsearch/reference/master/search-vector-tile-api.html#search-vector-tile-api-request-body) and conditionally add just the `query` property.

I'll improve this tomorrow, thanks!

Comment (jsanz) [reply index.js]: In [7fb3ee0](https://github.com/thomasneirynck/mvt_sample/pull/5/commits/7fb3ee0df0326ed0767a20024c74bf744eaf9673) I've moved the query parameters to the `body` and added a explicit query to retrieve everything if there's nothing to search.

File: index.js [hunk 2]
```diff
@@ -31,21 +31,32 @@ const server = http.createServer(async function (request, response) {
...
         try {
             const tile = await client.searchMvt({
                 index: params.index,
                 field: params.geometry,
                 zoom: parseInt(params.z),
                 x: parseInt(params.x),
                 y: parseInt(params.y),
-                exact_bounds: true,
-                extent: 4096,
                 grid_agg: params.renderMethod === 'grid' ? 'geotile' : 'geohex',
```
Comment (nreese) [inline index.js]: Can you also move grid_agg and track_total_hits to body please

File: index.js (hunk 1 above)
Comment (jsanz) [reply index.js]: Yep, done, sorry I missed those two from the docs.

File: index.js (hunk 2 above)
Comment (jsanz) [reply index.js]: Yep, done, sorry I missed those two from the docs.

**Review by nreese — APPROVED** [review]
LGTM

**Pull request merged**